### PR TITLE
Add support for SDU specific SSH keys

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Provide a way to use unique SSH key/credentials for an SDU. Currently all SDUs in an environment share the same SSH key/credentials from keyvault.

## Solution
Add an use_local_keyvault_for_secrets option in the options section. If set to true and the customer is not providing an existing sshkey the code will create an SSH key/credentials that will stored in the SDU keyvault - and all the VMs in the SDU will use it

## Tests
Add "use_local_keyvault_for_secrets" : true to the input json and deploy

## Notes
<Additional comments for the PR>

### CODE REVIEW

File: deploy/terraform/run/sap_system/variables_local.tf [hunk 1]
```diff
@@ -116,7 +116,7 @@ locals {
   landscape_tfstate_key        = try(var.landscape_tfstate_key, "")
 
   // Retrieve the arm_id of deployer's Key Vault from deployer's terraform.tfstate
-  deployer_key_vault_arm_id = try(data.terraform_remote_state.deployer.outputs.deployer_kv_user_arm_id, "")
+  deployer_key_vault_arm_id = try(var.key_vault.kv_spn_id, try(data.terraform_remote_state.deployer.outputs.deployer_kv_user_arm_id, ""))
```
Comment (rukawata) [inline deploy/terraform/run/sap_system/variables_local.tf:119]: This is to support bring your own deployer kv, should be a separate PR.

File: deploy/terraform/run/sap_system/sapsystem_full.json [hunk 2]
```diff
@@ -106,7 +105,15 @@
   "options": {
     "enable_secure_transfer": true,
     "enable_prometheus": true,
-    "resource_offset" : 0,
-    "disk_encryption_set_id" : ""
+    "resource_offset": 0,
+    "disk_encryption_set_id": "",
+    "use_local_keyvault_for_secrets" : false
```
Comment (rukawata) [inline deploy/terraform/run/sap_system/sapsystem_full.json:110]: Two comments on this variable:
1. Better put it under `sshkey` rather than `options`.
2. Use `enable_secrets_per_sid`, or `enable_secrets_per_sid`. `use_local_keyvault_for_secrets` seems confusing.

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_global.tf [hunk 3]
```diff
@@ -4,3 +4,7 @@ variable "infrastructure" {}
 variable "options" {}
 variable "ssh-timeout" {}
 variable "sshkey" {}
+variable "key_vault" {
+  description = "The user brings existing Azure Key Vaults"
+  default     = ""
+}
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_global.tf:10]: nit: newline.

**Review by rukawata — COMMENTED** [review]
Thanks. Some comments based on first look.

File: deploy/terraform/run/sap_system/sapsystem_full.json (hunk 2 above)
Comment (KimForss) [reply deploy/terraform/run/sap_system/sapsystem_full.json:110]: How about unique_credentials_per_sid or sid_specific_credentials?

**Comment by KimForss** [comment]
Replaced with https://github.com/Azure/sap-hana/pull/1029
